xFAANG/askql
# feat(prettier-plugin-askscript): intro

Closes #44 

To make sure that formatted source is correct, we now format files in memory before parsing them in unit tests. This is why some `.askcode` and `.askjson` files may be different.

This PR also includes some simplifications to parser that were needed:

- removal of function signature - variable definition is all we need. Semantically, this is now the simplest recursive function definition:
```askscript
const x = fun { x() }
```
- support for hex escape sequence (eg. `\xAA`) - required by [jesc](https://github.com/mathiasbynens/jsesc) which I used for formatting Unicode characters
- more flexible grammar for types: `id / functionCall`

**TEST PLAN**

1. Check out this PR
2. Install dependencies `npm install` and rebuild the project `npm run build`

3. a) Create formatted copies (`*.ask.formatted.ask`) all .ask files in the repo:

```
npm test
```

3. b) See the formatted output for a given ASK file:

```
ASK_PRINT_SOURCE=1 npm test src/askscript/__tests__/09-operators/operators-03-chained.ask
```
